jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorization.java

```diff
@@ -34,9 +46,66 @@
    * @return an {@link Authorization} with a {@code Bearer} token
    */
   public static Authorization fromBearerToken(String token) {
-    return new Authorization("Bearer", token);
+    return new Authorization("Bearer", token, decodeTokenRepositoryGrants(token));
   }
 
+  /**
+   * Decode the <a href="https://docs.docker.com/registry/spec/auth/jwt/">Docker Registry v2 Bearer
+   * Token</a> to list the granted repositories with their levels of access.
+   *
+   * @param token a Docker Registry Bearer Token
+   * @return a mapping of repository to granted access scopes, or {@code null} if there is no JWT
+   *     the token is not a Docker Registry Bearer Token
+   */
+  @VisibleForTesting
+  @Nullable
+  static Multimap<String, String> decodeTokenRepositoryGrants(String token) {
+    // Docker Registry Bearer Tokens are based on JWT.  The payload looks like:
+    // {
+    //   "access":[{"type":"repository","name":"repository/name","actions":["pull"]}],
+    //   "aud":"registry.docker.io",
+    //   "iss":"auth.docker.io",
+    //   "exp":999,
+    //   "iat":999,
+    //   "jti":"zzzz",
+    //   "nbf":999,
+    //   "sub":"e3ae001d-xxx"
+    // }
+    try {
+      DecodedJWT jwt = JWT.decode(token);
+      // Make sure they look like valid access claims.
+      JsonNode[] accessClaims = jwt.getClaim("access").asArray(JsonNode.class);
+      if (accessClaims == null
+          || !Stream.of(accessClaims).allMatch(Authorization::isValidAccessClaim)) {
+        return null;
+      }
+      return Stream.of(accessClaims)
+          .filter(n -> "repository".equals(n.get("type").asText()))
```

Comment:
`access` or `claim` instead of `n`? I didn't know it meant "Node" until I saw code below.